sisyphus/hash.py

```diff
@@ -44,6 +44,16 @@ def get_object_state(obj):
     Comment: Maybe obj.__reduce__() is a better idea? is it stable for hashing?
     """
 
+    # Note: sis_hash_helper does not call get_object_state in these cases.
+    # However, other code might (e.g. extract_paths),
+    # so we keep consistent to the behavior of sis_hash_helper.
+    if obj is None:
+        return None
+    if isinstance(obj, (bool, int, float, complex, str)):
```

Comment:
if `obj` is of type `np.float` then it is an instance of `float` so `get_object_state` simply returns it.

But `sis_hash_helper(obj)` checks for `type(obj) in (int, float, bool, str, complex):`, which is False and therefore we end up in the else case `byte_list.append(sis_hash_helper(get_object_state(obj)))` which leads to an infinite recursion.

Reply:
I pushed some change for that which makes this check more consistent. Can you check again?

Reply:
Whoops, sorry, wait, I think I messed sth up...

Reply:
Note, `np.float` is a bad example: This will actually break depending on your Numpy version. In (very old) Numpy versions, yes, `np.float` is not `float` but derived from `float`. However, in newer Numpy versions, `np.float` is just an alias to `float`.

> DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
> Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

So it means, if you really used `np.float` in your setup, the hash will break when you update Numpy...

A better example is a namedtuple. And here I accidentally also broke some hash now, but this is fixed now. Specifically, due to the `type(obj) in (tuple, list)` check, which was False, it also falls back to the `get_object_state` logic for namedtuples, so it's important that `get_object_state` behaves the same as before for namedtuples. This is what I do now.